File: src/main/scala/it/mighe/ssbi/instructions/DecrementValueInstruction.java
package it.mighe.ssbi.instructions;

import it.mighe.ssbi.Instruction;
import it.mighe.ssbi.Tape;

public class DecrementValueInstruction extends Instruction {

    public int execute(Tape tape, int programCounter) {
        tape.decrement();
        return programCounter + 1;
    }
}

File: src/main/scala/it/mighe/ssbi/instructions/DecrementValueInstruction.scala
package it.mighe.ssbi.instructions

import it.mighe.ssbi.Instruction
import it.mighe.ssbi.Tape

class DecrementValueInstruction extends Instruction {
  def execute(tape: Tape, programCounter: Int): Int = {
    tape.decrement()
    programCounter + 1
  }

  override def execute(tape: Tape): Instruction = {
    tape.decrement()
    next
  }
}